experimental/crabbake/src/bakeable.rs

```diff
@@ -0,0 +1,69 @@
+extern crate alloc;
+use alloc::borrow::Cow;
+use proc_macro2::TokenStream as TokenStream2;
+use quote::quote;
+
+pub trait Bakeable {
+    fn bake(&self) -> TokenStream2;
+}
+
+impl<'a> Bakeable for Cow<'a, str> {
+    fn bake(&self) -> TokenStream2 {
+        quote! {
+            alloc::borrow::Cow::Borrowed(#self)
+        }
+    }
+}
+
+impl Bakeable for u8 {
+    fn bake(&self) -> TokenStream2 {
+        quote! {
+            #self
+        }
+    }
+}
+
+impl<T, const N: usize> Bakeable for [T; N]
+where
+    T: Bakeable,
+{
+    fn bake(&self) -> TokenStream2 {
+        let data = self.iter().fold(TokenStream2::new(), |mut acc, value| {
+            let value = value.bake();
+            acc.extend(core::iter::once(quote! { #value, }));
+            acc
+        });
+
+        quote! {
+            &[#data]
```

Comment:
quote supports macro-style list interpolation, so you can write something like `&[#(data),*]` and not have to use the for loop above